UIforETW/TraceLoggingSupport.cpp

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright 2017 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "stdafx.h"
+#include "TraceLoggingSupport.h"
+#include "Utility.h"
+
+#include <bcrypt.h>
+
+#include <vector>
+#include <array>
+
+#ifdef _DEBUG
+#define new DEBUG_NEW
+#endif
+
+#pragma comment(lib, "Bcrypt.lib")
+
+static std::system_error MakeSystemErrorFromWin32Error(DWORD errorCode, const char* context)
```

Comment:
Where are these exceptions documented? It's not possible for xperf.exe to throw an exception back to UIforETW.